include/forcing/GridDataSelector.hpp

```diff
@@ -0,0 +1,300 @@
+        const auto ncells  = (row_max - row_min) * (col_max - col_min);
+
+        cells_.reserve(ncells);
+        for (auto row = row_min; row < row_max; row++) {
+            for (auto col = col_min; col < col_max; col++) {
+                cells_.emplace_back(Cell{/*x=*/col, /*y=*/row, /*z=*/0UL, /*value=*/NAN});
+            }
+        }
+    }
+
+    GridDataSelector() noexcept = default;
+
+    virtual ~GridDataSelector() = default;
+
+    time_t& initial_time() noexcept {
+        return config_.init_time;
+    }
+
+    time_t initial_time() const noexcept {
+        return config_.init_time;
+    }
+
+    long& duration() noexcept {
+        return config_.duration_seconds;
+    }
+
+    long duration() const noexcept {
+        return config_.duration_seconds;
+    }
+
+    std::string& variable() noexcept {
+        return config_.variable_name;
+    }
+
+    const std::string& variable() const noexcept {
+        return config_.variable_name;
+    }
+
+    std::string& units() noexcept {
+        return config_.variable_units;
+    }
+
+    const std::string& units() const noexcept {
+        return config_.variable_units;
+    }
+
+    boost::span<Cell> cells() noexcept {
+        return cells_;
+    }
+
+    boost::span<const Cell> cells() const noexcept {
+        return cells_;
+    }
+
+  private:
+    //! Returns true if point is inside polygon or on its boundary.
+    //! @note may not be needed
+    static bool intersects_(const geojson::coordinate_t& point, const geojson::polygon_t& polygon) {
+        const auto& boundary = polygon.outer();
+        const auto px        = point.get<0>();
+        const auto py        = point.get<1>();
+        size_t intersects    = 0;
+
+        for (size_t i = 0; i < boundary.size() - 1; i++) {
+            const auto& current_bpoint = boundary[i];
+            const auto& next_bpoint    = boundary[i + 1];
+            const double ix            = current_bpoint.get<0>();
+            const double iy            = current_bpoint.get<1>();
+            const double jx            = next_bpoint.get<0>();
+            const double jy            = next_bpoint.get<1>();
+            const double idx           = px - ix;
+            const double idy           = py - iy;
+            const double jdx           = px - jx;
+            const double jdy           = py - jy;
+            const double crossing      = ((idx - jdx) * idy) - (idx * (idy - jdy));
+
+            if (crossing == 0 && idx * jdx <= 0 and idy * jdy <= 0) {
+                return true;
+            }
+
+            if ((idy >= 0 && jdy < 0) || (jdy >= 0 && idy < 0)) {
+                if (crossing > 0) {
+                    intersects++;
+                } else if (crossing < 0) {
+                    intersects--;
+                }
+            }
+        }
+
+        return intersects != 0;
+    }
+
+    static Extent bounding_box_(const geojson::polygon_t& polygon) {
```

Comment:
It doesn't look like this gets used either.

As above, maybe this can be gotten from Boost Geometry. Otherwise, it seems like a standalone utility function, not something specific to this class

Reply:
Yeah you're correct, this and the above function were both initially for the boundary-based constructor. Boost.Geometry definitely has these, so I'm not sure why I sought to implement them 🤦🏾 I probably was hyper-fixating on then by mistake